services/databaseEncryptionService.ts

```diff
@@ -90,7 +107,11 @@ export async function initializeEncryption(): Promise<void> {
       if (error instanceof EncryptionError && error.code === ERROR_CODES.KEY_NOT_FOUND) {
         keyCache = await createNewKey();
       } else {
-        throw classifyError(error);
+        if (error instanceof EncryptionError) throw error;
+        if (isCancellationError(error)) {
+          throw new EncryptionError(ERROR_CODES.USER_CANCELLED, 'Authentication was cancelled');
+        }
```

Comment:
Correct me if I'm wrong, but if the first `if` is `true`, then so is the second one.